woocommerce/woocommerce
# E2E test script scaffold

### Changes proposed in this Pull Request:

This PR adds an E2E test installer to the `e2e-environment` and test install scaffolding to `e2e-core-tests`.

### How to test the changes in this Pull Request:

1. Run `cd plugins/woocommerce`
2. Run `pnpm install`
3. Rename the `tests` folder: `mv tests tests-x`
4. Run `pnpx wc-e2e install`
5. This should create 
- `tests/e2e/config/jest.config.js`
- `tests/e2e/config/jest.setup.js`
- `tests/e2e/docker/initialize.sh`
6. Rerun step #4
7. This should prompt for overwriting each of the above files
8. Run `pnpx wc-e2e install @woocommerce/e2e-core-tests`
9. This should create
- `tests/e2e/docker/woocommerce.e2e-core-tests.sh`
- `tests/e2e/config/default-woocommerce.e2e-core-tests.json`
- `tests/e2e/specs` folder structure
10. Review the created files
11. Run E2E tests
12. Run `pnpx wc-e2e uninstall @woocommerce/e2e-core-tests`
13. This should confirm the deletion of all files created in step #8
14. Delete the `tests` folder
15. Restore original tests folder: `mv tests-x tests`
